### DISCUSSION TITLE
How to access values passed into the `tpl` function

### DISCUSSION THREAD

**Discussion by armenr** [helmfile/helmfile#793] [Questions]
## Example

In my `helmfile.yaml`, I have this line: 

```
{{- $helperValues := tpl (readFile "../../some/relative/path/_helper_values.yaml.gotmpl") (dict "someValue" $tenantPath) | fromYaml -}}
```

In `_helper_values.yaml.gotmpl` I have:

```
renderMe: {{ .someValue }}
```

## Error

error during helmfile.yaml.gotmpl.part.0 parsing: template: stringTemplate:4:3: executing "stringTemplate" at <tpl (readFile "./_release-templates.yaml.gotmpl") .>: error calling tpl: template: stringTemplate:1:23: executing "stringTemplate" at <tpl (readFile "../components/_helper_values.yaml.gotmpl") .>: error calling tpl: template: stringTemplate:13:12: executing "stringTemplate" at <.someValue>: can't evaluate field someValue in type *state.EnvironmentTemplateData

## Question(s)

1. What is the correct way to access "someValue" that I passed in to the tpl function from inside the _helper_values.yaml.gotmpl?
2. **How can I merge the current context in my `helmfile.yaml` `(referenced with) .` WITH additional values I pass in?**

**Comment by armenr** [comment]
...anyone? @mumoshu ?

**Comment by yxxhero** [comment]
@armenr please give a minimal example. thanks very much.

**Comment by jouve** [comment]
I can't reproduce :
``` 
...
    values:
      - elastic-operator.yaml
      {{- $tenantPath := "bar" }}
      - {{ tpl (readFile "_helper_values.yaml.gotmpl") (dict "someValue" $tenantPath) }}
``` 
helmfile build | yq '.releases[0].values'
``` 
- elastic-operator.yaml
- renderMe: bar
```
